<launch>

    <param name="origin_lat" value="36.162263"/>
    <param name="origin_lon" value="-96.835559"/>
    <param name="origin_alt" value="315"/>
    <param name="dim" value="3"/>
    <param name="minX" value="-50"/>
    <param name="maxX" value="200"/>
    <param name="minY" value="-300"/>
    <param name="maxY" value="300"/>
    <param name="minZ" value="-20"/>
    <param name="maxZ" value="200"/>
    <param name="res" value="5"/>

    <arg name="As_neighbors" value="B_C"/>
    <arg name="Bs_neighbors" value="A_C"/>
    <arg name="Cs_neighbors" value="A_B"/>
    <!--<arg name="Ds_neighbors" value="A_C"/>-->
    <!--<arg name="Es_neighbors" value="B_C"/>-->

    <group ns="UAV">
        <node name="A" pkg="himher_ros" type="quad.py" respawn="false" output="screen">
            <param name="name" value="A"/>
            <param name="type" value="quad"/>
            <param name="neighbors" value="$(arg As_neighbors)"/>
            <param name="intent" value="unexplored_tempchange_boundary_avoidcollision"/>
        </node>
        <node name="B" pkg="himher_ros" type="quad.py" respawn="false" output="screen">
            <param name="name" value="B"/>
            <param name="type" value="quad"/>
            <param name="neighbors" value="$(arg Bs_neighbors)"/>
            <param name="intent" value="unexplored_tempchange_boundary_avoidcollision"/>
        </node>
        <node name="C" pkg="himher_ros" type="quad.py" respawn="false" output="screen">
            <param name="name" value="C"/>
            <param name="type" value="quad"/>
            <param name="neighbors" value="$(arg Cs_neighbors)"/>
            <param name="intent" value="unexplored_tempchange_boundary_avoidcollision"/>
        </node>
        <!--<node name="D" pkg="himher_ros" type="plane.py" respawn="false" output="screen">-->
            <!--<param name="name" value="D"/>-->
            <!--<param name="neighbors" value="$(arg Ds_neighbors)"/>-->
            <!--<param name="intent" value="unexplored_tempchange_boundary_avoidcollision"/>-->
        <!--</node>-->
        <!--<node name="E" pkg="himher_ros" type="plane.py" respawn="false" output="screen">-->
            <!--<param name="name" value="D"/>-->
            <!--<param name="neighbors" value="$(arg Ds_neighbors)"/>-->
            <!--<param name="intent" value="unexplored_tempchange_boundary_avoidcollision"/>-->
        <!--</node>-->

        <!--<node name="heart_bit_fly" pkg="himher_ros" type="heart_bit_fly.py" output="screen"></node>-->
    </group>

    <group ns="Visual">
        <node name="A" pkg="himher_ros" type="viz2D.py" respawn="false" output="screen">
            <param name="name" value="A"/>
        </node>
        <node name="B" pkg="himher_ros" type="viz2D.py" respawn="false" output="screen">
            <param name="name" value="B"/>
        </node>
        <node name="C" pkg="himher_ros" type="viz2D.py" respawn="false" output="screen">
            <param name="name" value="C"/>
        </node>
    </group>

</launch>
